### PR TITLE
feat: support test.each, test.todo etc.

### PR DESCRIPTION
**Pull request by jembezmamy**
This MR adds support for [all the test methods provided by QUnit](https://qunitjs.com/api/QUnit/test.each/):

- `test.if`
- `test.only`
- `test.skip`
- `test.todo`
- `test.each`
- `test.if.each`
- `test.only.each`
- `test.skip.each`
- `test.todo.each`

To allow setting `maxRuns` only once globally, a new argument was added to `setup`:

```javascript
// sets maxRuns to 3 for all methods
const retry = setup(QUnit.test, 3)
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
   "scripts": {
     "test": "pnpm test:smoke lint && pnpm test:js && pnpm test:smoke",
     "test:smoke": "testem ci",
-    "test:js": "qunit",
+    "test:js": "qunit test/retry.js && qunit test/retry-only.js",
```
Comment (mrloop) [inline package.json:29]: Is this needed, `qunit` by default should run all tests matching `test/**/*.js`

File: main.js [hunk 2]
```diff
@@ -1,7 +1,36 @@
 import Retry from './src/retry.js'
 
-export default function setup (testFn) {
-  return function retry (name, callback, maxRuns = 2) {
-    return new Retry(name, callback, maxRuns, testFn)
+export default function setup (testFn, defaultMaxRuns = 2) {
```
Comment (mrloop) [inline main.js:3]: :tada: this looks good, could you mention it in the docs. And the other functions :pray:

File: test/retry-only.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+const setup = require('../index.js')
```
Comment (mrloop) [inline test/retry-only.js:1]: Just wondering what the reason is for new test files for `retry.only` but not other functions?

File: package.json (hunk 1 above)
Comment (jembezmamy) [reply package.json:29]: Yes, it's necessary! We need to separate tests using `retry.only`, if we run them together with other tests, all the rest will be skipped. Is there a better way to isolate these two runs?

File: test/retry-only.js (hunk 3 above)
Comment (jembezmamy) [reply test/retry-only.js:1]: Maybe we could split the test file then, it's getting bigger and bigger. I isolated `retry.only` so it doesn't affect other tests

File: package.json (hunk 1 above)
Comment (mrloop) [reply package.json:29]: Ha, of course!

File: main.js (hunk 2 above)
Comment (jembezmamy) [reply main.js:3]: Updated!

Comment (mrloop) [reply main.js:3]: thanks

**Pull request merged**